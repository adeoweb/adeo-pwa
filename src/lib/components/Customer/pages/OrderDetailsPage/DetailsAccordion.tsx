import React, { FunctionComponent, useState } from 'react';
import { Accordion, Card } from 'react-bootstrap';
import ChevronIcon from 'src/lib/assets/icons/ChevronIcon';

interface IDetailsAccordionProps {
    startExpanded?: boolean;
    title: string;
    contentContainerClass?: string;
    lockToggle?: boolean;
}

const DetailsAccordion: FunctionComponent<IDetailsAccordionProps> = ({
    startExpanded = false,
    children,
    title,
    contentContainerClass = '',
    lockToggle
}) => {
    const [toggleStatus, setToggleStatus] = useState(startExpanded);

    const handleToggleClick = () => {
        if (!lockToggle) {
            setToggleStatus(!toggleStatus);
        }
    };

    return (
        <Accordion
            activeKey={toggleStatus ? 'details' : undefined}
            className={`customer-order-accordion ${
                toggleStatus ? 'customer-order-accordion-open' : ''
            }`}
        >
            <Card>
                <Accordion.Toggle
                    onClick={handleToggleClick}
                    eventKey={'details'}
                    className="customer-order-accordion-toggle"
                >
                    <div>{title}</div>
                    {!lockToggle && (
                        <ChevronIcon svgClass="customer-order-accordion-chevron" />
                    )}
                </Accordion.Toggle>
                <Accordion.Collapse
                    eventKey={'details'}
                    className="customer-order-accordion-collapse"
                >
                    <div
                        className={`customer-order-accordion-content ${contentContainerClass}`}
                    >
                        {children}
                    </div>
                </Accordion.Collapse>
            </Card>
        </Accordion>
    );
};

export default DetailsAccordion;
