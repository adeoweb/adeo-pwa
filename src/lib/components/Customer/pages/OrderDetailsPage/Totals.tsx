import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';
import { Price } from '@magento/peregrine';

interface ITotalsProps {
    grandTotal: number;
    subtotal: number;
    shippingAmount: number;
    taxAmount: number;
}

const Totals: FunctionComponent<ITotalsProps> = ({
    grandTotal,
    subtotal,
    shippingAmount,
    taxAmount
}) => {
    const { t } = useTranslation();

    return (
        <div className="customer-order-totals">
            <div className="customer-order-totals-container">
                <h4 className="customer-order-totals-title">
                    {t('Order information')}
                </h4>
                <div className="customer-order-totals-info">
                    <span>{t('Intermediate')}</span>
                    <span>
                        <Price
                            value={subtotal}
                            // TODO: get currency code from backend
                            currencyCode={'EUR'}
                        />
                    </span>
                </div>
                <div className="customer-order-totals-info">
                    <span>{t('TAX')}</span>
                    <span>
                        <Price
                            value={taxAmount}
                            // TODO: get currency code from backend
                            currencyCode={'EUR'}
                        />
                    </span>
                </div>
                <div className="customer-order-totals-info customer-order-totals-info-last">
                    <span>{t('Delivery')}</span>
                    <span>
                        <Price
                            value={shippingAmount}
                            // TODO: get currency code from backend
                            currencyCode={'EUR'}
                        />
                    </span>
                </div>
                <div className="customer-order-totals-grand">
                    <span>{t('Pay total')}</span>
                    <span>
                        <Price
                            value={grandTotal}
                            // TODO: get currency code from backend
                            currencyCode={'EUR'}
                        />
                    </span>
                </div>
            </div>
        </div>
    );
};

export default Totals;
