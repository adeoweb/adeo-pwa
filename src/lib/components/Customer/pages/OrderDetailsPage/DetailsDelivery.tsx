import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';
import { useUserContext } from 'src/peregrine/lib/context/adeoweb/user';
import DetailsAccordion from 'src/lib/components/Customer/pages/OrderDetailsPage/DetailsAccordion';
import { TOrderShippingAddress } from 'src/lib/types/graphql/Customer';

interface IDetailsDeliveryProps {
    shippingAddress: TOrderShippingAddress;
}

const DetailsDelivery: FunctionComponent<IDetailsDeliveryProps> = ({
    shippingAddress
}) => {
    const { t } = useTranslation();
    const [{ currentUser }] = useUserContext();

    if (!shippingAddress) {
        return null;
    }

    const {
        firstname,
        lastname,
        telephone,
        customer_notes: customerNotes,
        street: customerStreet,
        city: customerCity,
        postcode: customerPostcode
    } = shippingAddress;

    // TODO: get customer email for order from backend
    const email = (currentUser && currentUser.email) || null;

    return (
        <DetailsAccordion
            title={t('Delivery information')}
            contentContainerClass="customer-order-delivery"
        >
            <div className="customer-order-delivery-recipient">
                <div className="customer-order-delivery-info-block">
                    <div className="customer-order-font-description">
                        {`${t('First Name')}, ${t('Last Name')}`}
                    </div>
                    <div className="customer-order-font-info">{`${firstname} ${lastname}`}</div>
                </div>
                <div className="customer-order-delivery-info-block">
                    <div className="customer-order-font-description">
                        {t('Email ')}
                    </div>
                    <div className="customer-order-font-info">{email}</div>
                </div>
                <div className="customer-order-delivery-info-block">
                    <div className="customer-order-font-description">
                        {t('Phone Number')}
                    </div>
                    <div className="customer-order-font-info">{telephone}</div>
                </div>
            </div>
            <div>
                {/* TODO: uncomment when readdy */}
                {/*<div className="customer-order-font-emphasised">*/}
                {/*    /!* TODO: add courier logo and tittle *!/*/}
                {/*    Courier title*/}
                {/*</div>*/}
                <div className="customer-order-delivery-location">
                    <div className="customer-order-delivery-info-block">
                        <div className="customer-order-font-description">
                            {t('Address')}
                        </div>
                        <div className="customer-order-font-info">
                            {`${customerStreet}, ${customerCity}, ${customerPostcode}`}
                        </div>
                    </div>
                    <div className="customer-order-delivery-info-block">
                        <div className="customer-order-font-description">
                            {t('Phone Number')}
                        </div>
                        {customerNotes && (
                            <div className="customer-order-font-info">
                                {customerNotes}
                            </div>
                        )}
                    </div>
                </div>
            </div>
        </DetailsAccordion>
    );
};

export default DetailsDelivery;
