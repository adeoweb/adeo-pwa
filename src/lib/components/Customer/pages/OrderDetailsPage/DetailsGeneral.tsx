import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';

import PdfIcons from 'src/lib/assets/icons/PdfIcons';
import DetailsAccordion from 'src/lib/components/Customer/pages/OrderDetailsPage/DetailsAccordion';
import { TCustomerOrder } from 'src/lib/types/graphql/Customer';

interface IDetailsGeneralProps {
    order: TCustomerOrder;
}

const DetailsGeneral: FunctionComponent<IDetailsGeneralProps> = ({ order }) => {
    const { t } = useTranslation('order');

    const { created_at: createdAt, order_number: orderNumber, status } = order;

    return (
        <DetailsAccordion
            title={t('Order information')}
            startExpanded={true}
            contentContainerClass="customer-order-general"
            lockToggle={true}
        >
            <div className="customer-order-general-block">
                <div className="customer-order-font-description">
                    {t('Order No.')}
                </div>
                <div className="customer-order-font-info">{orderNumber}</div>
            </div>
            <div className="customer-order-general-block">
                <div className="customer-order-font-description">
                    {t('Status')}
                </div>
                <div className="customer-order-font-info">{status}</div>
            </div>
            <div className="customer-order-general-block">
                <div className="customer-order-font-description">
                    {t('Invoice')}
                </div>
                {/* TODO: add link to invoice when backend is done */}
                {/*<Link to="/" className="customer-order-font-info">*/}
                {/*    <PdfIcons />*/}
                {/*    {orderNumber}.pdf*/}
                {/*</Link>*/}
                <div className="customer-order-font-info">
                    <PdfIcons />
                    {orderNumber}.pdf
                </div>
            </div>
            <div className="customer-order-general-block">
                <div className="customer-order-font-description">
                    {t('Date of order')}
                </div>
                <div className="customer-order-font-info">{createdAt}</div>
            </div>
        </DetailsAccordion>
    );
};

export default DetailsGeneral;
