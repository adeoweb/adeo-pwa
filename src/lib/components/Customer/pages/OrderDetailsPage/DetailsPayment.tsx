import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';

import DetailsAccordion from 'src/lib/components/Customer/pages/OrderDetailsPage/DetailsAccordion';
// import swedbankLogo from 'src/lib/assets/images/swedbank_logo.png';
import { TOrderBillingAddress } from 'src/lib/types/graphql/Customer';
import { useUserContext } from 'src/peregrine/lib/context/adeoweb/user';

interface IDetailsPaymentProps {
    billingAddress: TOrderBillingAddress;
}

const DetailsPayment: FunctionComponent<IDetailsPaymentProps> = ({
    billingAddress
}) => {
    const { t } = useTranslation();
    const [{ currentUser }] = useUserContext();

    const { firstname, lastname, telephone, street, city, postcode } =
        billingAddress;

    // TODO: get customer email for order from backend
    const email = (currentUser && currentUser.email) || null;

    return (
        <DetailsAccordion
            title={t('Payment information')}
            contentContainerClass="customer-order-payment"
        >
            <div className="customer-order-font-emphasised">
                {t('Electronic banking and other methods')}
            </div>
            {/* TODO: uncomment when ready */}
            {/*<img*/}
            {/*    // TODO: get logo based on backend data*/}
            {/*    src={swedbankLogo}*/}
            {/*    // TODO: add logo title*/}
            {/*    alt=""*/}
            {/*    className="customer-order-payment-logo"*/}
            {/*/>*/}
            <div className="customer-order-payment-customer">
                <div className="customer-order-payment-info-block">
                    <div className="customer-order-font-description">
                        {`${t('First Name')}, ${t('Last Name')}`}
                    </div>
                    <div className="customer-order-font-info">{`${firstname} ${lastname}`}</div>
                </div>
                <div className="customer-order-payment-info-block">
                    <div className="customer-order-font-description">
                        {t('Email ')}
                    </div>
                    <div className="customer-order-font-info">{email}</div>
                </div>
                <div className="customer-order-payment-info-block">
                    <div className="customer-order-font-description">
                        {t('Phone Number')}
                    </div>
                    <div className="customer-order-font-info">{telephone}</div>
                </div>
                <div className="customer-order-payment-info-block">
                    <div className="customer-order-font-description">
                        {t('Address')}
                    </div>
                    <div className="customer-order-font-info">
                        {`${street}, ${city}, ${postcode}`}
                    </div>
                </div>
            </div>
        </DetailsAccordion>
    );
};

export default DetailsPayment;
