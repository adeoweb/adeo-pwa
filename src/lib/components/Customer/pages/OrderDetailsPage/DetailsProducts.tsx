import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';

import { Price } from '@magento/peregrine';

import DetailsAccordion from 'src/lib/components/Customer/pages/OrderDetailsPage/DetailsAccordion';
import { TOrderItem } from 'src/lib/types/graphql/Customer';

interface IDetailsProductsProps {
    items: TOrderItem[];
}

const DetailsProducts: FunctionComponent<IDetailsProductsProps> = ({
    items
}) => {
    const { t } = useTranslation(['order', 'product']);

    if (!Array.isArray(items) || !items.length) {
        return null;
    }

    return (
        <DetailsAccordion
            title={`${t('order:Goods')} (${items.length})`}
            contentContainerClass="customer-order-products"
        >
            {items.map(({ name, price_incl_tax: priceInclTax, qty, sku }) => (
                <div className="customer-order-product" key={sku}>
                    <div className="customer-order-product-details-block">
                        {/* TODO: get image url and label from backend */}
                        {/* TODO: uncomment when ready */}
                        {/*<Image src={''} alt={''} width={80} height={75} />*/}
                        <div>
                            <h3 className="customer-order-product-title">
                                {name}
                            </h3>
                            <table>
                                <tbody>
                                    <tr>
                                        <th>{t('product:Product code')}</th>
                                        <td>{sku}</td>
                                    </tr>
                                    {/* TODO: uncomment when ready */}
                                    {/*<tr>*/}
                                    {/*    <th>{t('Unit')}</th>*/}
                                    {/*    /!* TODO: get units from backend *!/*/}
                                    {/*    <td>matas</td>*/}
                                    {/*</tr>*/}
                                </tbody>
                            </table>
                        </div>
                    </div>
                    <div className="customer-order-product-order-block">
                        <div className="customer-order-product-order-info">
                            <div className="customer-order-product-description">
                                {t('product:Quantity')}
                            </div>
                            <div className="customer-order-product-value">
                                {qty}
                            </div>
                        </div>
                        <div className="customer-order-product-order-info">
                            <div className="customer-order-product-description">
                                {t('product:Price with tax')}
                            </div>
                            <div className="customer-order-product-value customer-order-product-price">
                                <Price
                                    value={priceInclTax?.value}
                                    // TODO: get currency code from backend
                                    currencyCode={priceInclTax?.currency}
                                />
                            </div>
                        </div>
                    </div>
                </div>
            ))}
        </DetailsAccordion>
    );
};

export default DetailsProducts;
