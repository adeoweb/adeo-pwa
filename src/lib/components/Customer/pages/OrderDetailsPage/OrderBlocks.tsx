import React, { Fragment, FunctionComponent } from 'react';
import DetailsGeneral from 'src/lib/components/Customer/pages/OrderDetailsPage/DetailsGeneral';
import DetailsProducts from 'src/lib/components/Customer/pages/OrderDetailsPage/DetailsProducts';
import DetailsDelivery from 'src/lib/components/Customer/pages/OrderDetailsPage/DetailsDelivery';
import DetailsPayment from 'src/lib/components/Customer/pages/OrderDetailsPage/DetailsPayment';
import Totals from 'src/lib/components/Customer/pages/OrderDetailsPage/Totals';
import { TCustomerOrder } from 'src/lib/types/graphql/Customer';

interface IOrderBloksProps {
    order: TCustomerOrder;
}

const OrderBlocks: FunctionComponent<IOrderBloksProps> = ({ order }) => {
    const {
        items,
        shipping_address: shippingAddress,
        billing_address: billingAddress,
        grand_total: grandTotal,
        subtotal,
        shipping_amount: shippingAmount,
        tax_amount: taxAmount
    } = order;

    return (
        <Fragment>
            <DetailsGeneral order={order} />
            <DetailsProducts items={items} />
            <DetailsDelivery shippingAddress={shippingAddress} />
            <DetailsPayment billingAddress={billingAddress} />
            <Totals
                grandTotal={grandTotal}
                subtotal={subtotal}
                shippingAmount={shippingAmount}
                taxAmount={taxAmount}
            />
        </Fragment>
    );
};

export default OrderBlocks;
