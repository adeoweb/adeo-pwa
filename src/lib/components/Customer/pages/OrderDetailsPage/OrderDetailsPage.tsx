import React, { FunctionComponent, useEffect } from 'react';
import { useTranslation } from 'react-i18next';

import { getSearchParam } from '@magento/peregrine/lib/hooks/useSearchParam';

import BackButton from 'src/lib/components/BackButton';
import { CustomerRoutes } from 'src/lib/components/Customer/CustomerRoutes';
import OrderBlocks from 'src/lib/components/Customer/pages/OrderDetailsPage/OrderBlocks';
import LoadingIndicator from 'src/lib/components/LoadingIndicator';
import MessageType from 'src/lib/constants/message';
import { useHistory, useLocation } from 'src/lib/drivers';
import GET_CUSTOMER_ORDER_DETAILS from 'src/lib/queries/getCustomerOrderDetails.graphql';
import { useCustomerOrders } from 'src/peregrine/lib/talons/adeoweb/Customer/useCustomerOrders';
import { useMessageCard } from 'src/peregrine/lib/talons/adeoweb/MessageCard/useMessageCard';

const OrderDetailsPage: FunctionComponent = () => {
    const { t } = useTranslation('order');
    const location = useLocation();
    const history = useHistory();
    const { addMessage } = useMessageCard();

    // TODO: replace query to get only specific order.

    const { loadingOrders, ordersError, orders } = useCustomerOrders({
        query: GET_CUSTOMER_ORDER_DETAILS
    });

    const id = Number(getSearchParam('id', location));

    if (!id) {
        history.push(CustomerRoutes.orders.url);
    }

    useEffect(() => {
        if (ordersError) {
            addMessage({
                type: MessageType.ERROR,
                message: String(ordersError)
            });
        }
    }, [addMessage, ordersError]);

    if (loadingOrders) {
        return <LoadingIndicator />;
    }

    const [order] = orders.filter(({ id: orderId }) => orderId === id);

    const orderNumber = (order && order.order_number) || null;

    return (
        <div>
            <BackButton url={CustomerRoutes.orders.url} />
            <h3 className="customer-content-title">
                {`${t('Order')} ${t('no.')} ${orderNumber || id}`}
            </h3>
            {order ? (
                <OrderBlocks order={order} />
            ) : (
                <div>{t('Order with given id does not exist')}</div>
            )}
        </div>
    );
};

export default OrderDetailsPage;
