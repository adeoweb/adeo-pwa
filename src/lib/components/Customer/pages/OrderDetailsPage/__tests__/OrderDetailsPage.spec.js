import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { createTestInstance } from '@magento/peregrine';

import OrderDetailsPage from 'src/lib/components/Customer/pages/OrderDetailsPage';

jest.mock('src/lib/drivers', () => {
    const useLocation = jest.fn(() => ({
        search: '?id=1'
    }));

    const useHistory = jest.fn(() => []);

    const Link = () => null;

    return { useLocation, useHistory, Link };
});

jest.mock('src/peregrine/lib/talons/adeoweb/Customer/useCustomerOrders', () => {
    const useCustomerOrders = jest.fn(() => ({
        orders: [
            {
                created_at: '2222-22-22',
                grand_total: 99,
                id: 1,
                order_number: 123,
                status: 'status',
                __typename: 'CustomerOrder',
                items: [
                    {
                        name: 'Įtampos stabilizatorius 7805 TO220',
                        price_incl_tax: 0.39,
                        qty: 1,
                        sku: '23188',
                        __typename: 'OrderItem'
                    },
                    {
                        name: 'Įtampos stabilizatorius 7905 TO220',
                        price_incl_tax: 0.59,
                        qty: 1,
                        sku: '23187',
                        __typename: 'OrderItem'
                    },
                    {
                        name: 'Klijų lazdelės Ø11mm 30cm, balti',
                        price_incl_tax: 0.55,
                        qty: 1,
                        sku: '23161',
                        __typename: 'OrderItem'
                    }
                ],
                shipping_address: {
                    firstname: 'M',
                    lastname: 'K',
                    telephone: '+37061234567',
                    customer_notes: 'comment',
                    street: ['Address'],
                    city: 'City',
                    postcode: '10001',
                    department: null,
                    __typename: 'OrderShippingAddress'
                },
                billing_address: {
                    firstname: 'M',
                    lastname: 'K',
                    telephone: '+37061234567',
                    street: ['Address'],
                    city: 'City',
                    postcode: '10001',
                    __typename: 'OrderBillingAddress'
                },
                subtotal: 1.53,
                shipping_amount: 0,
                tax_amount: 0
            }
        ]
    }));

    return { useCustomerOrders };
});

jest.mock('src/peregrine/lib/talons/adeoweb/MessageCard/useMessageCard', () => {
    const useMessageCard = jest.fn(() => ({
        addMessage: jest.fn(() => {})
    }));

    return { useMessageCard };
});

jest.mock('src/peregrine/lib/context/adeoweb/user', () => {
    const api = {
        currentUser: {
            email: 'name@domain.com'
        }
    };
    const useUserContext = jest.fn(() => [api]);

    return { useUserContext };
});

test('renders correctly', async () => {
    const instance = createTestInstance(
        <Router>
            <OrderDetailsPage />
        </Router>
    );

    expect(instance.toJSON()).toMatchSnapshot();
});
