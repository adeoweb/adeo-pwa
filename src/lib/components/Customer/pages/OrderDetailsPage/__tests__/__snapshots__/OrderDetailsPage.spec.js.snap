// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3
    className="customer-content-title"
  >
    Order no. 123
  </h3>
  <div
    className="customer-order-accordion customer-order-accordion-open accordion"
  >
    <div
      className="card"
    >
      <button
        className="customer-order-accordion-toggle"
        onClick={[Function]}
        type="button"
      >
        <div>
          Order information
        </div>
      </button>
      <div
        aria-expanded={null}
        className="customer-order-accordion-collapse collapse show"
      >
        <div
          className="customer-order-accordion-content customer-order-general"
        >
          <div
            className="customer-order-general-block"
          >
            <div
              className="customer-order-font-description"
            >
              Order No.
            </div>
            <div
              className="customer-order-font-info"
            >
              123
            </div>
          </div>
          <div
            className="customer-order-general-block"
          >
            <div
              className="customer-order-font-description"
            >
              Status
            </div>
            <div
              className="customer-order-font-info"
            >
              status
            </div>
          </div>
          <div
            className="customer-order-general-block"
          >
            <div
              className="customer-order-font-description"
            >
              Invoice
            </div>
            <div
              className="customer-order-font-info"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clipRule="evenodd"
                  d="M6 2H14L20 8V20C20 21.1 19.1 22 18 22H5.99C4.89 22 4 21.1 4 20L4.01 4C4.01 2.9 4.9 2 6 2ZM13 3.5V9H18.5L13 3.5ZM7.16406 15V17H6.01562V11.3125H7.95312C8.51562 11.3125 8.96354 11.487 9.29687 11.8359C9.63281 12.1849 9.80078 12.638 9.80078 13.1953C9.80078 13.7526 9.63541 14.1927 9.30469 14.5156C8.97396 14.8385 8.51562 15 7.92969 15H7.16406ZM7.16406 14.043H7.95312C8.17187 14.043 8.34114 13.9714 8.46094 13.8281C8.58073 13.6849 8.64062 13.4766 8.64062 13.2031C8.64062 12.9193 8.57943 12.694 8.45703 12.5273C8.33463 12.3581 8.17057 12.2721 7.96484 12.2695H7.16406V14.043ZM10.5703 17V11.3125H12.0742C12.7383 11.3125 13.2669 11.5234 13.6602 11.9453C14.056 12.3672 14.2578 12.9453 14.2656 13.6797V14.6016C14.2656 15.349 14.0677 15.9362 13.6719 16.3633C13.2786 16.7878 12.7357 17 12.043 17H10.5703ZM11.7187 12.2695V16.0469H12.0625C12.4453 16.0469 12.7148 15.9466 12.8711 15.7461C13.0273 15.543 13.1094 15.194 13.1172 14.6992V13.7109C13.1172 13.1797 13.043 12.8099 12.8945 12.6016C12.7461 12.3906 12.4935 12.2799 12.1367 12.2695H11.7187ZM16.2422 14.6797H18.0273V13.7266H16.2422V12.2695H18.2422V11.3125H15.0937V17H16.2422V14.6797Z"
                  fill="#F15642"
                  fillRule="evenodd"
                />
              </svg>
              123
              .pdf
            </div>
          </div>
          <div
            className="customer-order-general-block"
          >
            <div
              className="customer-order-font-description"
            >
              Date of order
            </div>
            <div
              className="customer-order-font-info"
            >
              2222-22-22
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="customer-order-accordion  accordion"
  >
    <div
      className="card"
    >
      <button
        className="customer-order-accordion-toggle"
        onClick={[Function]}
        type="button"
      >
        <div>
          order:Goods (3)
        </div>
        <svg
          className="customer-order-accordion-chevron"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M9.99984 6L8.58984 7.41L13.1698 12L8.58984 16.59L9.99984 18L15.9998 12L9.99984 6Z"
            fill="#2E2E2E"
            fillRule="evenodd"
          />
        </svg>
      </button>
      <div
        aria-expanded={null}
        className="customer-order-accordion-collapse collapse"
      >
        <div
          className="customer-order-accordion-content customer-order-products"
        >
          <div
            className="customer-order-product"
          >
            <div
              className="customer-order-product-details-block"
            >
              <div>
                <h3
                  className="customer-order-product-title"
                >
                  Įtampos stabilizatorius 7805 TO220
                </h3>
                <table>
                  <tbody>
                    <tr>
                      <th>
                        product:Product code
                      </th>
                      <td>
                        23188
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
            <div
              className="customer-order-product-order-block"
            >
              <div
                className="customer-order-product-order-info"
              >
                <div
                  className="customer-order-product-description"
                >
                  product:Quantity
                </div>
                <div
                  className="customer-order-product-value"
                >
                  1
                </div>
              </div>
              <div
                className="customer-order-product-order-info"
              >
                <div
                  className="customer-order-product-description"
                >
                  product:Price with tax
                </div>
                <div
                  className="customer-order-product-value customer-order-product-price"
                >
                  <div />
                </div>
              </div>
            </div>
          </div>
          <div
            className="customer-order-product"
          >
            <div
              className="customer-order-product-details-block"
            >
              <div>
                <h3
                  className="customer-order-product-title"
                >
                  Įtampos stabilizatorius 7905 TO220
                </h3>
                <table>
                  <tbody>
                    <tr>
                      <th>
                        product:Product code
                      </th>
                      <td>
                        23187
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
            <div
              className="customer-order-product-order-block"
            >
              <div
                className="customer-order-product-order-info"
              >
                <div
                  className="customer-order-product-description"
                >
                  product:Quantity
                </div>
                <div
                  className="customer-order-product-value"
                >
                  1
                </div>
              </div>
              <div
                className="customer-order-product-order-info"
              >
                <div
                  className="customer-order-product-description"
                >
                  product:Price with tax
                </div>
                <div
                  className="customer-order-product-value customer-order-product-price"
                >
                  <div />
                </div>
              </div>
            </div>
          </div>
          <div
            className="customer-order-product"
          >
            <div
              className="customer-order-product-details-block"
            >
              <div>
                <h3
                  className="customer-order-product-title"
                >
                  Klijų lazdelės Ø11mm 30cm, balti
                </h3>
                <table>
                  <tbody>
                    <tr>
                      <th>
                        product:Product code
                      </th>
                      <td>
                        23161
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
            <div
              className="customer-order-product-order-block"
            >
              <div
                className="customer-order-product-order-info"
              >
                <div
                  className="customer-order-product-description"
                >
                  product:Quantity
                </div>
                <div
                  className="customer-order-product-value"
                >
                  1
                </div>
              </div>
              <div
                className="customer-order-product-order-info"
              >
                <div
                  className="customer-order-product-description"
                >
                  product:Price with tax
                </div>
                <div
                  className="customer-order-product-value customer-order-product-price"
                >
                  <div />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="customer-order-accordion  accordion"
  >
    <div
      className="card"
    >
      <button
        className="customer-order-accordion-toggle"
        onClick={[Function]}
        type="button"
      >
        <div>
          order:Delivery Information
        </div>
        <svg
          className="customer-order-accordion-chevron"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M9.99984 6L8.58984 7.41L13.1698 12L8.58984 16.59L9.99984 18L15.9998 12L9.99984 6Z"
            fill="#2E2E2E"
            fillRule="evenodd"
          />
        </svg>
      </button>
      <div
        aria-expanded={null}
        className="customer-order-accordion-collapse collapse"
      >
        <div
          className="customer-order-accordion-content customer-order-delivery"
        >
          <div
            className="customer-order-delivery-recipient"
          >
            <div
              className="customer-order-delivery-info-block"
            >
              <div
                className="customer-order-font-description"
              >
                customer:First Name, customer:Last Name
              </div>
              <div
                className="customer-order-font-info"
              >
                M K
              </div>
            </div>
            <div
              className="customer-order-delivery-info-block"
            >
              <div
                className="customer-order-font-description"
              >
                customer:Email
              </div>
              <div
                className="customer-order-font-info"
              >
                name@domain.com
              </div>
            </div>
            <div
              className="customer-order-delivery-info-block"
            >
              <div
                className="customer-order-font-description"
              >
                customer:Phone Number
              </div>
              <div
                className="customer-order-font-info"
              >
                +37061234567
              </div>
            </div>
          </div>
          <div>
            <div
              className="customer-order-delivery-location"
            >
              <div
                className="customer-order-delivery-info-block"
              >
                <div
                  className="customer-order-font-description"
                >
                  customer:Address
                </div>
                <div
                  className="customer-order-font-info"
                >
                  Address, City, 10001
                </div>
              </div>
              <div
                className="customer-order-delivery-info-block"
              >
                <div
                  className="customer-order-font-description"
                >
                  customer:Phone Number
                </div>
                <div
                  className="customer-order-font-info"
                >
                  comment
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="customer-order-accordion  accordion"
  >
    <div
      className="card"
    >
      <button
        className="customer-order-accordion-toggle"
        onClick={[Function]}
        type="button"
      >
        <div>
          order:Payment Information
        </div>
        <svg
          className="customer-order-accordion-chevron"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M9.99984 6L8.58984 7.41L13.1698 12L8.58984 16.59L9.99984 18L15.9998 12L9.99984 6Z"
            fill="#2E2E2E"
            fillRule="evenodd"
          />
        </svg>
      </button>
      <div
        aria-expanded={null}
        className="customer-order-accordion-collapse collapse"
      >
        <div
          className="customer-order-accordion-content customer-order-payment"
        >
          <div
            className="customer-order-font-emphasised"
          >
            order:Electronic banking and other methods
          </div>
          <div
            className="customer-order-payment-customer"
          >
            <div
              className="customer-order-payment-info-block"
            >
              <div
                className="customer-order-font-description"
              >
                customer:First Name, customer:Last Name
              </div>
              <div
                className="customer-order-font-info"
              >
                M K
              </div>
            </div>
            <div
              className="customer-order-payment-info-block"
            >
              <div
                className="customer-order-font-description"
              >
                customer:Email
              </div>
              <div
                className="customer-order-font-info"
              >
                name@domain.com
              </div>
            </div>
            <div
              className="customer-order-payment-info-block"
            >
              <div
                className="customer-order-font-description"
              >
                customer:Phone Number
              </div>
              <div
                className="customer-order-font-info"
              >
                +37061234567
              </div>
            </div>
            <div
              className="customer-order-payment-info-block"
            >
              <div
                className="customer-order-font-description"
              >
                customer:Address
              </div>
              <div
                className="customer-order-font-info"
              >
                Address, City, 10001
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="customer-order-totals"
  >
    <div
      className="customer-order-totals-container"
    >
      <h4
        className="customer-order-totals-title"
      >
        Order information
      </h4>
      <div
        className="customer-order-totals-info"
      >
        <span>
          Intermediate
        </span>
        <span>
          <div />
        </span>
      </div>
      <div
        className="customer-order-totals-info"
      >
        <span>
          TAX
        </span>
        <span>
          <div />
        </span>
      </div>
      <div
        className="customer-order-totals-info customer-order-totals-info-last"
      >
        <span>
          Delivery
        </span>
        <span>
          <div />
        </span>
      </div>
      <div
        className="customer-order-totals-grand"
      >
        <span>
          Pay total
        </span>
        <span>
          <div />
        </span>
      </div>
    </div>
  </div>
</div>
`;
